packages/core/contracts-ovm/insured-bridge/implementation/OVM_BridgeDepositBox.sol

```diff
@@ -1,47 +1,249 @@
 // SPDX-License-Identifier: AGPL-3.0-only
 pragma solidity >=0.7.6;
 
+import "./OVM_Testable.sol"; //TODO: replace this with the normal UMA Testable once we can use 0.8 solidity.
+
 import { OVM_CrossDomainEnabled } from "@eth-optimism/contracts/libraries/bridge/OVM_CrossDomainEnabled.sol";
+import { Lib_PredeployAddresses } from "@eth-optimism/contracts/libraries/constants/Lib_PredeployAddresses.sol";
+
+// Define some interfaces and helper libraries. This is temporary until we can bump the solidity
+// version in these contracts to 0.8.x and import the rest of these libs from other UMA contracts in the repo.
+library TokenHelper {
+    function safeTransferFrom(
+        address token,
+        address from,
+        address to,
+        uint256 value
+    ) internal {
+        // bytes4(keccak256(bytes('transferFrom(address,address,uint256)')));
+        (bool success, bytes memory data) = token.call(abi.encodeWithSelector(0x23b872dd, from, to, value));
+        require(
+            success && (data.length == 0 || abi.decode(data, (bool))),
+            "TokenHelper::transferFrom: transferFrom failed"
+        );
+    }
+}
+
+interface TokenLike {
+    function balanceOf(address guy) external returns (uint256 wad);
+}
+
+interface StandardBridgeLike {
+    function withdrawTo(
+        address _l2Token,
+        address _to,
+        uint256 _amount,
+        uint32 _l1Gas,
+        bytes calldata _data
+    ) external;
+}
+
+/**
+ * @title OVM Bridge Deposit Box.
+ * @notice Accepts deposits on Optimism L2 to relay to Ethereum L1 as part of the UMA insured relayer system.
+ */
+
+contract OVM_BridgeDepositBox is OVM_CrossDomainEnabled, OVM_Testable {
+    /*************************************
+     *  OVM DEPOSIT BOX DATA STRUCTURES  *
+     *************************************/
+
+    address public bridgeRouter;
 
-contract OVM_BridgeDepositBox is OVM_CrossDomainEnabled {
-    address public l1Owner;
+    bool public depositsEnabled = true;
 
-    event Deposit(
-        uint256 transferId,
+    // Track the total number of deposits. Used as a unique identifier for bridged transfers.
+    uint256 public numberOfDeposits;
+
+    struct WhitelistedToken {
```

Comment:
nit: add comment explaining that this struct represents the mappings to a whitelisted L2 token. FYI in the BridgeRouter I name this struct `L1TokenRelationships`, if you like that name then you could name this `L2TokenRelationships` but I can see why you'd prefer `WhitelistedToken`

Reply:
I have a comment on the mapping. is that not sufficient? I will change to `L2TokenRelationships` for consistency.

Reply:
Sure